search.go

```diff
@@ -38,32 +38,38 @@ func NewSearchRequest() *SearchRequest {
 	return &SearchRequest{Size: -1}
 }
 
-// Source translates the SearchRequest into the shape expected by OpenSearch.
-func (r *SearchRequest) Source() (map[string]any, error) {
+// MarshalJSON marshals the SearchRequest into the JSON shape expected by OpenSearch.
+// Implemented to match the [json.Marshaler] interface.
+func (r *SearchRequest) MarshalJSON() ([]byte, error) {
 	source := make(map[string]any)
 	if r.Query != nil {
-		src, err := r.Query.Source()
-		if err != nil {
-			return nil, err
+		queryJSON, jErr := r.Query.ToOpenSearchJSON()
+		if jErr != nil {
+			return nil, jErr
 		}
 
-		source["query"] = src
+		source["query"] = json.RawMessage(queryJSON)
```

Comment:
Nice "trick" here with using `json.RawMessage` to hold the intermediate encoded data.